python/pyarrow/scalar.pxi

```diff
@@ -64,6 +64,13 @@ cdef class ArrayValue(Scalar):
         else:
             return super(Scalar, self).__repr__()
 
+    def __str__(self):
+        self._check_null()
```

Comment:
I believe `_check_null` is no longer needed, @kszucs is that right?

Reply:
@wesm after merging https://github.com/apache/arrow/pull/2132 `_check_null` won't be needed